test/test_cabinet.py

```diff
@@ -151,6 +152,98 @@ def test_read_create(self):
         self.assertDictEqual(data, result)
         client.delete_bucket(Bucket=bucket)
 
+    def test_read_create_with_default_region(self):
+        client = boto3.client('s3', 'us-east-2')
+        bucket = 'mock-bucket'
+        client.create_bucket(
+            Bucket=bucket,
+            CreateBucketConfiguration={'LocationConstraint': 'us-east-2'}
+        )
+        protocol, filename = 's3', f'{bucket}/test.yml'
+        data = {'I': {'am': ['nested', 1, 'object', None]}}
+        cabinets.create(f'{protocol}://{filename}', data)
+        result = cabinets.read(f'{protocol}://{filename}')
+        cabinets.delete(f'{protocol}://{filename}')
+        self.assertDictEqual(data, result)
+        client.delete_bucket(Bucket=bucket)
+
+    def test_read_create_with_different_region(self):
+        client = boto3.client('s3', 'us-east-2')
+        bucket = 'mock-bucket'
+        client.create_bucket(
+            Bucket=bucket,
+            CreateBucketConfiguration={'LocationConstraint': 'us-east-2'}
+        )
+        protocol, filename = 's3', f'{bucket}/test.yml'
+        data = {'I': {'am': ['nested', 1, 'object', None]}}
+        cabinets.create(f'{protocol}://{filename}', data)
+        result = cabinets.read(f'{protocol}://{filename}')
+        cabinets.delete(f'{protocol}://{filename}')
+        self.assertDictEqual(data, result)
+        client.delete_bucket(Bucket=bucket)
+
+
+@mock_s3
+@patch.dict(os.environ, {'AWS_ACCESS_KEY_ID': 'testing',
+                         'AWS_SECRET_ACCESS_KEY': 'testing',
+                         'AWS_SECURITY_TOKEN': 'testing',
+                         'AWS_SESSION_TOKEN': 'testing', })
+class TestS3CabinetWithRegion(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self._region = 'us-west-2'
+        S3Cabinet.set_configuration(region_name=self._region)
+
+    def test_set_configuration_region(self):
+        self.assertIsNotNone(S3Cabinet.client)
+        self.assertEqual(S3Cabinet.client.meta.region_name, self._region)
+
+    def test_read_create_s3_cabinet(self):
+        client = boto3.client('s3', region_name=self._region)
+        bucket = 'mock-bucket'
+        client.create_bucket(
+            Bucket=bucket,
+            CreateBucketConfiguration={'LocationConstraint': self._region}
+        )
+        filename = f'{bucket}/test.yml'
+        data = {'I': {'am': ['nested', 1, 'object', None]}}
+        S3Cabinet.create(f'{filename}', data)
+        result = S3Cabinet.read(f'{filename}')
+        S3Cabinet.delete(f'{filename}')
+        self.assertDictEqual(data, result)
+        client.delete_bucket(Bucket=bucket)
```

Comment:
Wondering if any of this can be moved into tear down method. I know there are a number of scenarios we're trying test and it may not be worth it, but maybe we can collect all the tests that need to delete a bucket in one class?